File: frontend/src/types/plan.ts
export interface DailyPlanTask {
  task_id: string
  predicted_start: string
  predicted_end: string
  priority_score: number
  title: string
  is_critical: boolean
  is_urgent: boolean
  action_plan?: string[] | null  // List of actionable steps to complete the task
  description?: string | null  // Original task description for context
}

export interface DailyPlan {
  id: string
  user_id: string
  plan_date: string
  tasks: DailyPlanTask[]
  energy_level: number | null
  status: string
  generated_at: string
  created_at: string
  updated_at: string
}

export interface EnergyLevel {
  id: string
  user_id: string
  date: string
  energy_level: number
  created_at: string
  updated_at: string
}

export interface Reminder {
  id: string
  title: string
  description: string | null
  start_time: string
  end_time: string
  is_all_day: boolean
  date: string
}

File: backend/app/models/plan.py
"""Daily Plan models"""
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime, date
from uuid import UUID


class DailyPlanTask(BaseModel):
    """Individual task in a daily plan"""
    task_id: UUID
    predicted_start: datetime
    predicted_end: datetime
    priority_score: float
    title: str
    is_critical: bool = False
    is_urgent: bool = False


class DailyPlan(BaseModel):
    """Complete daily plan"""
    id: Optional[UUID] = None
    user_id: UUID
    plan_date: date
    tasks: List[DailyPlanTask]
    energy_level: Optional[int] = None
    status: str = "active"
    generated_at: Optional[datetime] = None
    created_at: Optional[datetime] = None
    
    class Config:
        json_encoders = {
            datetime: lambda v: v.isoformat(),
            date: lambda v: v.isoformat(),
            UUID: lambda v: str(v)
        }


class PlanningContext(BaseModel):
    """Input context for planning"""
    raw_tasks: List[dict]  # List of RawTask dictionaries
    energy_level: int
    plan_date: date
    time_constraints: Optional[dict] = None  # Optional global time constraints
    
    class Config:
        json_encoders = {
            date: lambda v: v.isoformat(),
        }


class DailyPlanCreate(BaseModel):
    """Daily plan creation model"""
    user_id: UUID
    plan_date: date
    tasks: List[DailyPlanTask]
    energy_level: Optional[int] = None
    status: str = "active"


class DailyPlanResponse(BaseModel):
    """Daily plan response model"""
    id: UUID
    user_id: UUID
    plan_date: date
    tasks: List[dict]  # JSONB from database
    energy_level: Optional[int] = None
    status: str
    generated_at: datetime
    created_at: datetime
    updated_at: datetime

